uast/uast.go

```diff
@@ -152,8 +152,37 @@ const (
 	// in a certain scope.
 	ImportAlias
 
-	// TODO: arguments, return value, body, etc.
+	// TODO: argument type declarations, return value, body, etc.
+
+	// FunctionDeclaration is the parent node of all function or method declarations. It should have
+	// FunctionDeclarationName and FunctionDeclarationBody children and, if the function has formal arguments,
+	// FunctionDeclarationArgument children. Please note that some languages could support more than one
+	// FunctionDeclarationBody roles.
```

Comment:
`Please note that some languages could support more than one
FunctionDeclarationBody roles.`

I wouldn't add this here unless if have a good reason to do it.

Cases such as Elixir or Scala functions will require to carefully look at their AST to see how can we annotate them.